packages/experience/src/pages/Consent/index.tsx

```diff
@@ -1,54 +1,114 @@
-import { conditional } from '@silverhand/essentials';
-import { useEffect, useContext, useState } from 'react';
+import { type ConsentInfoResponse } from '@logto/schemas';
+import { useCallback, useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
 
-import PageContext from '@/Providers/PageContextProvider/PageContext';
-import { consent } from '@/apis/consent';
-import { LoadingIcon } from '@/components/LoadingLayer';
+import LandingPageLayout from '@/Layout/LandingPageLayout';
+import { consent, getConsentInfo } from '@/apis/consent';
+import Button from '@/components/Button';
+import TextLink from '@/components/TextLink';
 import useApi from '@/hooks/use-api';
 import useErrorHandler from '@/hooks/use-error-handler';
-import { getBrandingLogoUrl } from '@/utils/logo';
 
+import OrganizationSelector, { type Organization } from './OrganizationSelector';
+import ScopesListCard from './ScopesListCard';
+import UserProfile from './UserProfile';
 import * as styles from './index.module.scss';
 
 const Consent = () => {
-  const { experienceSettings, theme } = useContext(PageContext);
   const handleError = useErrorHandler();
   const asyncConsent = useApi(consent);
-  const { branding, color } = experienceSettings ?? {};
-  const brandingLogo = conditional(
-    branding &&
-      color &&
-      getBrandingLogoUrl({ theme, branding, isDarkModeEnabled: color.isDarkModeEnabled })
-  );
+  const { t } = useTranslation();
+
+  const [consentData, setConsentData] = useState<ConsentInfoResponse | undefined>();
+  const [selectedOrganization, setSelectedOrganization] = useState<Organization | undefined>();
+
+  const asyncGetConsentInfo = useApi(getConsentInfo);
+
+  const consentHandler = useCallback(async () => {
+    const [error, result] = await asyncConsent(selectedOrganization?.id);
 
-  const [loading, setLoading] = useState(true);
+    if (error) {
+      await handleError(error);
+
+      return;
+    }
+
+    if (result?.redirectTo) {
+      window.location.replace(result.redirectTo);
+    }
+  }, [asyncConsent, handleError, selectedOrganization?.id]);
 
   useEffect(() => {
-    (async () => {
-      const [error, result] = await asyncConsent();
-      setLoading(false);
+    const getConsentInfoHandler = async () => {
+      const [error, result] = await asyncGetConsentInfo();
 
       if (error) {
         await handleError(error);
 
         return;
       }
 
-      if (result?.redirectTo) {
-        window.location.replace(result.redirectTo);
-      }
-    })();
-  }, [asyncConsent, handleError]);
+      setConsentData(result);
+    };
+
+    void getConsentInfoHandler();
+  }, [asyncGetConsentInfo, handleError]);
+
+  // Init the default organization selection
+  useEffect(() => {
+    if (!consentData?.organizations?.length) {
+      return;
+    }
+
+    setSelectedOrganization(consentData.organizations[0]);
+  }, [consentData]);
```

Comment:
Seems this can be merged into the `useEffect` hook above? After `setConsentData`?

Reply:
updated